src/puppetlabs/trapperkeeper/services/authorization/authorization_service.clj

```diff
@@ -0,0 +1,23 @@
+(ns puppetlabs.trapperkeeper.services.authorization.authorization-service
+  (:require [puppetlabs.trapperkeeper.core :refer [defservice]]
+            [puppetlabs.trapperkeeper.authorization.ring-middleware :refer
+             [wrap-authorization-check]]
+            [puppetlabs.trapperkeeper.services.authorization.authorization-core :refer :all]))
+
+(defprotocol AuthorizationService
+  (wrap-handler [this handler]))
+
+(defservice authorization-service
+  AuthorizationService
+  [[:ConfigService get-in-config]]
+  (init
+    [this context]
+    (let [config (get-in-config [:authorization :rules])]
+      (validate-auth-config! config)
+      (assoc context :authorization {:rules (transform-config config)})))
+
+  (wrap-handler
+    [this handler]
+    (let [context (service-context this)
```

Comment:
@camlow325 @nwolfe Would appreciate your help with this one...  I'm at the point where I'm trying to finish the last item of TK-258, which is to actually implement the TK service function that accepts a ring request handler and returns a new handler wrapped with the authorization check.

I'm trying to obtain the service context using `service-context` as mentioned at https://github.com/puppetlabs/trapperkeeper/wiki/Defining-Services  but I keep running into the error:

```
CompilerException java.lang.RuntimeException: Unable to resolve symbol: service-context in this context, compiling:(src/puppetlabs/trapperkeeper/services/authorization/authorization_service.clj:21:19) 
```

Any ideas?  Looking at other uses, it seems to be correct so I think I'm definitely overlooking something.